ios/Nodes/REAParamNode.m

```diff
@@ -3,37 +3,35 @@
 #import "REANodesManager.h"
 
 @implementation REAParamNode {
-    NSString *_name;
-    NSMutableArray<REANodeID> *_argstack;
+  NSMutableArray<REANodeID> *_argstack;
 }
 
 - (instancetype)initWithID:(REANodeID)nodeID config:(NSDictionary<NSString *,id> *)config
 {
-    if ((self = [super initWithID:nodeID config:config])) {
-        _name = config[@"name"];
-        _argstack = [NSMutableArray<REANodeID> arrayWithCapacity:0];
-    }
-    return self;
+  if ((self = [super initWithID:nodeID config:config])) {
+    _argstack = [NSMutableArray<REANodeID> arrayWithCapacity:0];
+  }
+  return self;
 }
 
--(void) setValue:(NSNumber *)value {
-    REANode *node = [self.nodesManager findNodeByID:[_argstack lastObject]];
-    [(REAValueNode*)node setValue:value];
+- (void)setValue:(NSNumber *)value
+{
+  REANode *node = [self.nodesManager findNodeByID:[_argstack lastObject]];
+  [(REAValueNode*)node setValue:value];
 }
 
--(void) beginContext:(NSNumber*) ref {
-    [_argstack addObject:ref];
+- (void)beginContext:(NSNumber*) ref {
+  [_argstack addObject:ref];
 }
 
--(void) endContext {
-    [_argstack removeLastObject];
+- (void)endContext {
+  [_argstack removeLastObject];
 }
 
 
--(id) evaluate {
-    REANode * node = [self.nodesManager findNodeByID:[_argstack lastObject]];
-    return [node value];
+- (id)evaluate {
+  REANode * node = [self.nodesManager findNodeByID:[_argstack lastObject]];
```

Comment:
probably here we need to go back to the previous context (set `prevCallId` as callID) and after evaluation  revert this change.

Reply:
Just testing to see if the node exists..? :) Maybe not necessary?

Reply:
The problem still exist. The problem arises if we use the `set` node in different contexts (different call id) - then the memoized value will be set, but when someone tries to read it again the wrong memoized value will be read. 

I think that using the dictionary to store memoized values is the problem - it will eventually lead to situations where different values are stored for the same node - giving unpredictable results when reading nodes from different contexts.